mne_bids/path.py

```diff
@@ -626,6 +627,135 @@ def mkdir(self, exist_ok=True):
         self.directory.mkdir(parents=True, exist_ok=exist_ok)
         return self
 
+    @verbose
+    def rm(self, safe_remove=True, verbose=None):
+        """Safely delete a set of files inside BIDS dataset.
+
+        Deleting a scan that conforms to the bids-validator, will
+        delete both the row in the ``*scans.tsv`` file, and also
+        the corresponding sidecar files and the data file itself.
+
+        Deleting all files of a subject will update the
+        ``*participants.tsv`` file.
+
+
+        Parameters
+        ----------
+        safe_remove : bool
+            If ``False``, directly delete and update the files.
+            Otherwise, displays the list of operations planned
+            and asks for user confirmation before
+            executing them (default).
+        %(verbose)s
+
+        Returns
+        -------
+        self : BIDSPath
+            The BIDSPath object.
+        """
+        # only proceed if root is defined
+        if self.root is None:
+            raise RuntimeError("The root must not be None to remove files.")
+
+        # Planning:
+        paths_matched = self.match(ignore_json=False, check=self.check)
+        subjects = set()
+        paths_to_delete = []
```

Comment:
```suggestion
        paths_to_delete = list()
```

just a stylistic suggestion is to use `list()` instead of `[]` and `dict()` instead of `{}`. To not confuse between the two and also to be more explicit.